projects/plugins/jetpack/modules/sso.php

```diff
@@ -83,6 +88,52 @@ private function __construct() {
 		}
 	}
 
+	/**
+	 * Render the invitation email message.
+	 */
+	public function render_invitation_email_message() {
+		printf( '<div class="notice inline notice-large notice-info">%1s</div>', esc_html__( 'The user will be automatically invited to WordPress.com.', 'jetpack' ) );
+	}
+
+	/**
+	 * Intercept the core invitation email.
+	 *
+	 * @param bool    $notify Whether to notify the user or not.
+	 * @param WP_User $user   The user object.
+	 *
+	 * @return bool Whether to notify the user or not via core email.
+	 */
+	public function intercept_core_invitation_email( $notify, $user ) {
+		if ( ! $user instanceof WP_User ) {
+			return $user;
+		}
+
+		$email   = $user->user_email;
+		$role    = $user->roles[0];
```

Comment:
I think this is the right way to get the role, I checked other places in Jetpack and that's how it is done.

```
$roles         = new Roles();
$user_role = $roles->translate_user_to_role( $user );
```